airbyte_cdk/cli/airbyte_cdk/_secrets.py

```diff
@@ -131,25 +132,52 @@ def fetch(
     )
     # Fetch and write secrets
     secret_count = 0
+    failed_secrets = []
+    failed_secret_urls = []
+
     for secret in secrets:
         secret_file_path = _get_secret_filepath(
             secrets_dir=secrets_dir,
             secret=secret,
         )
-        _write_secret_file(
+        error = _write_secret_file(
             secret=secret,
             client=client,
             file_path=secret_file_path,
         )
-        click.echo(f"Secret written to: {secret_file_path.absolute()!s}", err=True)
-        secret_count += 1
 
-    if secret_count == 0:
+        if error:
+            secret_name = secret.name.split("/secrets/")[-1]  # Removes project prefix
+            failed_secrets.append(secret_name)
+            secret_url = f"https://console.cloud.google.com/security/secret-manager/secret/{secret_name}/versions?hl=en&project={gcp_project_id}"
```

Comment:
Make this URL concatenation into a helper function, taking `gcp_project_id` and `secret_name` as inputs. Then refactor the new exception class to take these inputs instead: `connector_name`, `gcp_project_id`, and `secret_names`. Then the exception class can call the same helper function to print URLs.